server/src/main/java/io/druid/server/ClientQuerySegmentWalker.java

```diff
@@ -91,37 +91,42 @@ public ClientQuerySegmentWalker(
   private <T> QueryRunner<T> makeRunner(Query<T> query, QueryRunner<T> baseClientRunner)
   {
     QueryToolChest<T, Query<T>> toolChest = warehouse.getToolChest(query);
+    return new ResultLevelCachingQueryRunner<>(makeRunner(query, baseClientRunner, toolChest),
```

Comment:
@drcrallen Does this look ok? I have refactored it a bit but it doesn't exactly follow the fluent style.
Incorporating ResultLevelCachingQueryRunner inside the FluentQueryRunnerBuilder is tricky because the query runner logic needs to keep track of cache value data before and after the baseRunner.run is invoked. Further this query runner is not accessible inside the fluent query runner builder, which makes it a bit more complex.
I have addressed your other comments as well.

Reply:
`io.druid.query.FluentQueryRunnerBuilder.FluentQueryRunner#emitCPUTimeMetric` use in the fluent query runner builder has a similar need. If you make a method in the fluent builder that takes the missing parameters (query, objectmapper, cache, cacheconfig) does it work?

Reply:
@drcrallen Tried that but `FluentQueryRunnerBuilder` which resides in `druid-processing` does not have access to `ResultLevelCachingQueryRunner` as it (along with most of the caching logic) resides in `druid-server` package. 
I attempted to refactor the caching logic into druid-processing, but there are several other dependencies that may have to be moved. I could work on a follow up improvement PR to investigate and perform the refactoring and hopefully make the `makeRunner` method cleaner.

Reply:
That must be quite frustrating >.<

Can you clarify this in an issue, and put the issue in a comment above this code.

That way anyone coming in later and wondering why it is like this can have a clear logic path for what needs to change to fix things up.

Reply:
@drcrallen I've made the requested changes.